flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/split/IcebergSourceSplit.java

```diff
@@ -157,21 +165,56 @@ byte[] serializeV2() throws IOException {
     return serializedBytesCache;
   }
 
+  private static void writeLongUTF(DataOutputSerializer out, String taskJson, int version)
```

Comment:
this method is inaccurate. `writeTaskJson` is probably more accurate.

Reply:
Agree, will push once all other comments are concluded with a change plan.